src/core/lib/surface/call.h

```diff
@@ -120,6 +120,9 @@ size_t grpc_call_get_initial_size_estimate();
 grpc_compression_algorithm grpc_call_compression_for_level(
     grpc_call* call, grpc_compression_level level);
 
+/* Did this client call receive a trailers-only response */
+bool grpc_call_is_trailers_only(const grpc_call* call);
```

Comment:
Reaching under the hood to an internal C-core API seems a little messy.  Is there any reason other than short-term expediency why we would want to go with this approach, rather than something like #13423 that actually exposes it in the C-core surface API?

This could be useful for wrapped languages as well; I don't think we've looked at their APIs yet to know whether they need it.  But even if it's used only by C++, it seems like it would be cleaner to go through the C-core API instead of reaching into its internals.

Reply:
So we already reach into core internals in numerous places in the C++ API (eg closures) because we had decided that the C++ API can be de-wrapped. This function would be easy enough to implement as core surface if the need arises, but I'm gonna cite YAGNI here and suggest that it shouldn't be moved to core surface until there's a blocking feature request for it from some other language that is not de-wrappable. (And there's also the practical issue that I won't be around to write or shepherd any new gRFC for a Core API extension.)

So I'm going to ask if I can push back against this review comment. I'm accepting the other one.

Reply:
That said, thanks for the review!

Reply:
Okay, I guess we can live with this for now.  Please add a TODO about this, though.

Reply:
Done, thanks! Adding it in your name since you'll make the call when it happens.